libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -415,15 +424,25 @@ std::optional<VotesBundle> VoteManager::getVotesBundleByRoundAndStep(uint64_t ro
       for (auto const& voted_value : found_step_it->second) {
         if (voted_value.second.first >= two_t_plus_one) {
           const auto voted_block_hash = voted_value.first;
-          auto it = voted_value.second.second.begin();
           size_t count = 0;
 
-          // Copy at least 2t+1 votes
-          while (count < two_t_plus_one) {
-            votes.emplace_back(it->second);
-            count += it->second->getWeight().value();
-            it++;
+          if (step == 3) {
+            // Collect all cert votes
```

Comment:
why do you need to collect all cert votes ? This method is used to "CHECK IF WE HAVE RECEIVED 2t+1 CERT VOTES FOR A BLOCK IN OUR CURRENT ROUND" - why would you need more than  `two_t_plus_one` votes ?

Reply:
In the previous approch, only collect 2t+1 cert votes by getVotesBundleByRoundAndStep that will save in period_data column in DB, other cert votes are not collected that will be droped.
Since we use previous cert votes as reward votes in proposal PBFT block. for example: 2t+1 is 3
node1 has cert votes v1, v2, v3, v4. If only save {v1, v2, v3} for finalized block B1. When other node propose B2 with reward votes {v1, v2, v3, v4}, node1 will not be able to get v4 again because of node1 has seen v4 already.
So the new approch need save all received cert votes for finalized block instead of only saving 2t+1